universum/analyzers/uncrustify.py

```diff
@@ -1,200 +1,196 @@
 import argparse
 import difflib
 import sys
-import os
+from os import environ
+from pathlib import Path
 
 import re
-import sh
+from typing import Callable, List, Optional, Tuple
 
 from . import utils
 
-# The maximum number of lines to write separate comments for
-# If exceeded, summarized comment will be provided instead
-MAX_LINES = 11
 
+def form_arguments_for_documentation() -> argparse.ArgumentParser:  # TODO: modify callers to remove wrapper
+    return _uncrustify_argument_parser()
+
+
+def main() -> int:
+    settings: argparse.Namespace = _uncrustify_argument_parser().parse_args()
+    target_folder: Path = Path.cwd().joinpath(settings.output_directory)
+    if not settings.cfg_file and 'UNCRUSTIFY_CONFIG' not in environ:
+        sys.stderr.write("Please specify the '--cfg_file' parameter "
+                         "or set an env. variable 'UNCRUSTIFY_CONFIG'")
+        return 2
+    wrapcolumn, tabsize = _get_wrapcolumn_tabsize(settings.cfg_file)
+    differ = difflib.HtmlDiff(tabsize=tabsize, wrapcolumn=wrapcolumn)
+
+    def write_html_diff_file(file: Path, src: List[str], target: List[str]) -> None:
+        # target_folder and differ used from outer scope
+        file_relative = file.relative_to(Path.cwd())
+        out_file_name: str = str(file_relative).replace('/', '_') + '.html'
+        with open(target_folder.joinpath(out_file_name), 'w') as out_file:
+            out_file.write(differ.make_file(src, target, context=False))
+
+    files: List[Tuple[Path, Path]] = []
+    try:
+        src_files: List[Path] = _get_src_files(settings.file_list, settings.file_list_config)
+        for pattern in settings.pattern_form:
+            regexp = re.compile(pattern)  # TODO: combine patterns via join
+            src_files = [file for file in src_files if regexp.match(str(file))]
+        for src_file in src_files:
+            src_file_relative = src_file.relative_to(Path.cwd())
+            target_file: Path = target_folder.joinpath(src_file_relative)
+            files.append((src_file, target_file))
+        if not files:
+            raise EnvironmentError("Please provide at least one file for analysis")
+    except EnvironmentError as e:
+        sys.stderr.write(str(e))
+        return 2
+
+    cmd = ["uncrustify", "-q", "-c", settings.cfg_file, "--prefix", settings.output_directory]
+    cmd.extend([str(path.relative_to(Path.cwd())) for path in src_files])
+
+    return utils.report_parsed_outcome(cmd,
+                                       lambda _: _uncrustify_output_parser(files, write_html_diff_file),
```

Comment:
Both the "write_html_diff_file" function and "_uncrustify_output_parser" function are used in this only place.
However, the former is local to the main function and is passed to the latter as a callback parameter.
As I see it, the only (technical) reason for this is using of "target_folder" and "differ" variables.
At the same time the lambda uses "files" variable.

Please consider to uniform the usage of variables. For instance, you may only access them in local function and pass to the global function as parameters when needed.